### PR TITLE
Added lifecycle methods, set_action_bar_button

### PR DESCRIPTION
**Pull request by jamonholmgren**
Added `on_attach`, `on_create`, `on_create_view`, `on_activity_created` (same as `on_load`), `on_start`, `on_resume`, `on_pause`, `on_stop`, `on_destroy_view`, `on_destroy`, `on_detach`.

Also added a preliminary version of `add_action_bar_button`. It's pretty rough, needs work, but does create a button.

### CODE REVIEW

File: lib/project/pro_motion/pm_screen_module.rb [hunk 1]
```diff
@@ -164,5 +164,31 @@ def action_bar
...
+      options[:show] ||= :always
+
+      # Should be something like Android::MenuItem::SHOW_AS_ACTION_IF_ROOM
+      show_as_action = 0 if options[:show] == :never
+      show_as_action = 1 if options[:show] == :if_room
+      show_as_action = 2 if options[:show] == :always
+      show_as_action = 4 if options[:show] == :with_text
+      show_as_action = 8 if options[:show] == :collapse
+
+      btn = self.activity.menu.add(options.fetch(:group, 0), options.fetch(:item_id, 0), options.fetch(:order, 0), options.fetch(:title, "Untitled"))
```
Comment (jamonholmgren) [inline lib/project/pro_motion/pm_screen_module.rb]: Will document and probably add a nicer DSL to this later.

File: lib/project/pro_motion/pm_single_fragment_activity.rb [hunk 2]
```diff
@@ -38,5 +38,10 @@ def set_fragment(fragment)
       fragmentManager.beginTransaction.add(@fragment_container.getId, fragment, fragment.class.to_s).commit
     end
 
+    def on_create_menu(menu)
+      @menu = menu
+      self.fragment.on_create_menu(menu) if self.fragment
+    end
```
Comment (jamonholmgren) [inline lib/project/pro_motion/pm_single_fragment_activity.rb]: Forwards on to the PMScreen so they can add buttons.

**Comment by GantMan** [comment]
SELF MERGER!!!!! :point_up:

**Comment by twerth** [comment]
Heh ^

**Comment by jamonholmgren** [comment]
Todd told me to!

**Comment by jamonholmgren** [comment]
There was a bug in it, I pushed directly to master, InfiniteRed-style.

**Comment by GantMan** [comment]
:P

**Pull request merged**